src/attributes.jl

```diff
@@ -250,7 +250,17 @@ end
 function Base.setindex!(attrdict::AttributeDict, val, name::AbstractString)
     if haskey(attrdict, name)
         # in case of an error, we write first to a temporary, then rename
-        _name = tempname()
+        max_tries = 100 # number of times to try making a new name
+        _name = nothing
+        for i in 1:max_tries
```

Comment:
Since we've moved to uuid, I really don't expect there to be a clash using uuid4 since the probability of collision is so incredibly low. Perhaps we could just remove this and simply this.